src/lib/edf_schedule.c

```diff
@@ -346,7 +346,7 @@ static void schedule_edf_task_normal(struct task *task, uint64_t start,
 	/* need to run scheduler if task not already running */
 	if (need_sched) {
 		/* rerun scheduler */
-		schedule();
+		schedule_edf();
```

Comment:
Can you state why in commit message, looks like an optimization or bug fix ?

Reply:
I've add information that is for optimization. In fact, it was my inattention during #1089 PR. We should not use generic functions in specific scheduler implementation.
Effect is the same, because ```schedule()``` will also invoke ```schedule_edf()``` function.